Mathlib/Data/Seq/Computation.lean

```diff
@@ -503,8 +503,8 @@ theorem length_thinkN (s : Computation α) [_h : Terminates s] (n) :
 
 theorem eq_thinkN {s : Computation α} {a n} (h : Results s a n) : s = thinkN (pure a) n := by
   revert s
-  induction n with | zero => _ | succ n IH => _
-  all_goals intro s; apply recOn s (fun a' => _) fun s => _ <;> intro a h
+  induction n with | zero => _ | succ n IH => _ <;>
+  intro s <;> apply recOn s (fun a' => _) fun s => _ <;> intro a h
```

Comment:
Could you use `x <;> (y; w; z)` here instead, to make the points at which goals bifurcate clearer? As is, you can't tell from the syntax that there is no further splitting.

Reply:
I modified this, I was not able to use `<;> [...]`, but I hope I made it clearer.

What is going on is that the first `intro <;> apply` is really an `intro; apply` for the initial 2 goals.  After the `apply` there are 4 goals left and the final `intro` applies to all.